assets/js/modules/adsense/components/dashboard/DashboardSummaryWidget.js

```diff
@@ -0,0 +1,179 @@
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+
+/**
+ * WordPress dependencies
+ */
+import { __, _x } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from '../../datastore/constants';
+import { STORE_NAME as CORE_USER } from '../../../../googlesitekit/datastore/user/constants';
+import { reduceAdSenseData } from '../../util';
+import { readableLargeNumber, extractForSparkline, getSiteKitAdminURL } from '../../../../util';
+import whenActive from '../../../../util/when-active';
+import PreviewBlock from '../../../../components/preview-block';
+import Layout from '../../../../components/layout/layout';
+import DataBlock from '../../../../components/data-block';
+import Sparkline from '../../../../components/sparkline';
+import getDataErrorComponent from '../../../../components/notifications/data-error';
+import getNoDataComponent from '../../../../components/notifications/nodata';
+
+const { useSelect } = Data;
+
+function DashboardSummaryWidget() {
+	const {
+		error,
+		loading,
+		today,
+		period,
+		daily,
+	} = useSelect( ( select ) => {
+		const store = select( STORE_NAME );
+		const metrics = [ 'EARNINGS', 'PAGE_VIEWS_RPM', 'IMPRESSIONS' ];
+
+		const format = 'YYYY-MM-DD';
+		const now = moment().format( format );
+		const lastMonth = moment().subtract( 1, 'month' ).format( format );
+
+		const todayArgs = {
+			startDate: now,
+			endDate: now,
+			metrics,
+		};
+
+		const periodArgs = {
+			dateRange: select( CORE_USER ).getDateRange(),
+			metrics,
+		};
+
+		const dailyArgs = {
+			startDate: lastMonth,
+			endDate: now,
+			metrics,
+			dimensions: [ 'DATE' ],
+		};
+
+		return {
+			error: store.getErrorForSelector( 'getReport', [ todayArgs ] ) || store.getErrorForSelector( 'getReport', [ periodArgs ] ) || store.getErrorForSelector( 'getReport', [ dailyArgs ] ),
+			loading: store.isResolving( 'getReport', [ todayArgs ] ) || store.isResolving( 'getReport', [ periodArgs ] ) || store.isResolving( 'getReport', [ dailyArgs ] ),
+			today: store.getReport( todayArgs ),
+			period: store.getReport( periodArgs ),
+			daily: store.getReport( dailyArgs ),
+		};
+	} );
+
+	if ( loading ) {
+		return <PreviewBlock width="100%" height="276px" />;
+	}
+
+	if ( error ) {
+		return getDataErrorComponent( 'adsense', error.message );
+	}
+
+	if ( ! today.totals && ! period.totals && ! daily.totals ) {
+		return getNoDataComponent( __( 'AdSense', 'google-site-kit' ) );
+	}
+
+	const processedData = reduceAdSenseData( daily.rows );
+	const href = getSiteKitAdminURL( 'googlesitekit-module-adsense', {} );
+
+	const currencyHeader = period.headers.find( ( header ) => null !== header.currency && 0 < header.currency.length );
+	const currencyCode = currencyHeader ? currencyHeader.currency : false;
+
+	return (
+		<Layout className="googlesitekit-dashboard-adsense-stats" fill>
```

Comment:
Let's not use this legacy component here, we should be either wrapping this in a `Widget` manually, or alternatively provide `wrapWidget: true` in the registration.

Reply:
Updated to use `Widget` to be able to pass the class.